pennylane/operation.py

```diff
@@ -695,6 +695,25 @@ def hash(self):
             )
         )
 
+    def __eq__(self, other):
+        warnings.warn(
+            "The behaviour of Operator.__eq__ will be updated soon. Currently, op1 == op2 is "
+            "True if op1 and op2 are the same object. Soon, op1 == op2 will be equivalent to "
+            "qml.equal(op1, op2). To continue using operator equality in its current state, "
+            "use 'op1 is op2'.",
+            UserWarning,
+        )
+        return super().__eq__(other)
+
+    def __hash__(self):
+        warnings.warn(
+            "The behaviour of Operator.__hash__ will be updated soon. Currently, Operator.__hash__ "
+            "is equivalent to object.__hash__. Soon, Operator.__hash__ will be determined by the "
+            "combined __hash__ of the name, wires, parameters and hyperparameters of the Operator.",
```

Comment:
I'm open to discuss for this hashing one, but could we also not mention the dunder here? E.g.,
> The behaviour of Operator hashing will be updated soon. Currently, each instance of Operator will
have a unique hash. Soon, an Operator's hash will be determined by the combined hashes of the name,
wires, parameters, and hyperparameters of the Operator. To continue using Operator hashing in its current state, use the wrapper (TODO).